Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -95,44 +89,121 @@ private void HandleShuttleMovement(SharedPlayerInputMoverComponent mover)
-            // inputs will do different things.
-            // TODO: Do that
-            float speedCap;
-            var angularSpeed = 0.075f;
-
             // ShuttleSystem has already worked out the ratio so we'll just multiply it back by the mass.
-            var movement = (mover.VelocityDir.walking + mover.VelocityDir.sprinting);
+            var movement = mover.VelocityDir.walking + mover.VelocityDir.sprinting;
+            var system = EntitySystem.Get<ThrusterSystem>();
+
+            if (movement.Length.Equals(0f))
+            {
+                // TODO: This visualization doesn't work with multiple pilots so need to address that somehow.
+                system.DisableAllThrustDirections(shuttleComponent);
+                return;
+            }
+
+            var speedCap = 0f;
 
             switch (shuttleComponent.Mode)
             {
                 case ShuttleMode.Docking:
-                    if (physicsComponent.LinearVelocity.LengthSquared == 0f)
+                    system.DisableAllThrustDirections(shuttleComponent);
+                    var dockDirection = movement.ToWorldAngle().GetDir();
+                    var dockFlag = dockDirection.AsFlag();
+
+                    // Won't just do cardinal directions.
+                    foreach (DirectionFlag dir in Enum.GetValues(typeof(DirectionFlag)))
                     {
-                        movement *= 5f;
-                    }
+                        // Brain no worky but I just want cardinals
+                        switch (dir)
+                        {
+                            case DirectionFlag.South:
+                            case DirectionFlag.East:
+                            case DirectionFlag.North:
+                            case DirectionFlag.West:
+                                break;
+                            default:
+                                continue;
+                        }
+
+                        if ((dir & dockFlag) == 0x0) continue;
 
-                    if (movement.Length != 0f)
-                        physicsComponent.ApplyLinearImpulse(physicsComponent.Owner.Transform.WorldRotation.RotateVec(movement) * shuttleComponent.SpeedMultipler * physicsComponent.Mass);
+                        system.EnableThrustDirection(shuttleComponent, dir);
+
+                        var index = (int) Math.Log2((int) dir);
+
+                        var dockSpeed = shuttleComponent.LinearThrusterImpulse[index];
+
+                        // TODO: Cvar for the speed drop
+                        dockSpeed /= 5f;
+
+                        if (physicsComponent.LinearVelocity.LengthSquared == 0f)
+                        {
+                            dockSpeed *= 5f;
+                        }
+
+                        var moveAngle = movement.ToWorldAngle();
+
+                        // Currently this is slow BUT we'd have a separate multiplier for docking and cruising or whatever.
```

Comment:
isnt this obsolete now?

Reply:
I mean it technically exists but we need an angular velocity cap cvar + a multiplier cvar at some stage.